File: RouterLib/src/main/java/com/kronos/router/BindRouter.java
package com.kronos.router;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by Leif Zhang on 2016/12/2.
 * Email leifzhanggithub@gmail.com
 */

@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.CLASS)
public @interface BindRouter {
    String[] urls();

    boolean isRunnable() default false;

    int weight() default 0;
}

File: RouterAnnotation/src/main/java/com/kronos/router/BindRouter.kt
package com.kronos.router

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import kotlin.reflect.KClass

/**
 * Created by Leif Zhang on 2016/12/2.
 * Email leifzhanggithub@gmail.com
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(RetentionPolicy.CLASS)
annotation class BindRouter(val urls: Array<String>, val weight: Int = 0, val interceptors: Array<KClass<*>> = []) {

}